src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -114,11 +121,27 @@ private void Initialize(Compilation comp, AnalyzerExecutor analyzerExecutor, Can
             }
         }
 
-        private Task ExecuteSyntaxTreeActions(CancellationToken cancellationToken)
+        public void StartCompleteAnalysis(ImmutableHashSet<SyntaxTree> syntaxTreesToSkip, CancellationToken cancellationToken)
+        {
+            if (_initializeTaskStarted && _primaryTask == null)
+            {
+                _primaryTask = Task.Run(async () =>
+                {
+                    await _initializeTask.ConfigureAwait(false);
+                    // Compilation start actions execute as part of the initialize task, not as part of the compilation events task,
+                    // so executing the syntax tree actions here puts them between compilation start actions and other actions.
+                    await ExecuteSyntaxTreeActions(_compilation.SyntaxTrees.Where((tree) => !syntaxTreesToSkip.Contains(tree)), cancellationToken).ConfigureAwait(false);
+                    await ProcessCompilationEventsAsync(true, cancellationToken).ConfigureAwait(false);
```

Comment:
Minor: Consider use of named arguments for Boolean arguments for better readability.